framework/base/Security.php

```diff
@@ -521,14 +521,16 @@ public function generateRandomKey($length = 32)
             if ($urandomFile !== null) {
                 // $length could be large so read using a loop.
                 $key = '';
+                $bytesLeft = $length;
                 do {
-                    $buffer = fread($urandomFile, $length);
+                    $buffer = fread($urandomFile, min($bytesLeft, 8092));
```

Comment:
right, thanks @tom--.

How do we proceed here?

This will read in chunks and is covered by the test:
```php
$buffer = fread($urandomFile, min($bytesLeft, 8192));
```

This will read in nearly all cases in on chunk but there is no test to verify it can handle the case where multiple chunks are read:
```php
$buffer = fread($urandomFile, $bytesLeft);
```
Still we have now reviewed it and know it works. which one is better?

performance is no difference as far as I see (tested with reading 50MB).